### PR TITLE
Userland: Fix some silly bugs

### PR DESCRIPTION
**Pull request by boricj**
`pkgsrc` support on SerenityOS has gone from "not working" to "fall flat on one's face at the first bootstrap step".

Let's fall flat on one's face at the second bootstrap step instead.

### CODE REVIEW

File: Userland/Utilities/pwd.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 ErrorOr<int> serenity_main(Main::Arguments)
 {
-    TRY(Core::System::pledge("stdio"));
+    TRY(Core::System::pledge("rpath stdio"));
```
Comment (IdanHo) [inline Userland/Utilities/pwd.cpp:14]: Commit message nitpick: mind being a bit more destrictive about what you're fixing? a-la "Add missing rpath pledge"

File: Userland/Libraries/LibCore/FilePermissionsMask.cpp [hunk 2]
```diff
@@ -48,47 +49,60 @@ ErrorOr<FilePermissionsMask> FilePermissionsMask::from_symbolic_notation(StringV
...
-                else
-                    return Error::from_string_literal("invalid operation: expected '+', '-' or '='"sv);
-
-                state = State::Mode;
+                state = State::Operation;
+                break;
             }
+        }
+            [[fallthrough]];
+
```
Comment (IdanHo) [inline Userland/Libraries/LibCore/FilePermissionsMask.cpp:79]: optional nitpick: having an empty line after a fallthrough looks wonky

File: Userland/Utilities/pwd.cpp (hunk 1 above)
Comment (boricj) [reply Userland/Utilities/pwd.cpp:14]: The commit message has been updated.

File: Userland/Libraries/LibCore/FilePermissionsMask.cpp (hunk 2 above)
Comment (boricj) [reply Userland/Libraries/LibCore/FilePermissionsMask.cpp:79]: Fixed.

File: Userland/Utilities/groups.cpp [hunk 3]
```diff
@@ -44,12 +44,12 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     args_parser.parse(arguments);
 
     if (usernames.is_empty()) {
-        auto account = TRY(Core::Account::from_uid(geteuid()));
+        auto account = TRY(Core::Account::from_uid(geteuid(), Core::Account::Read::PasswdOnly));
```
[the remark this answers is not in the document]
Comment (boricj) [reply Userland/Utilities/groups.cpp:47]: `groups` flat-out doesn't work with non-root users.

It seems SerenityOS could use more regression/POSIX conformance testing. I didn't encounter any of these issues last time, so these silly bugs have been introduced in the meantime.

**Comment by IdanHo** [comment]
@boricj You might have forgotten to push?

**Comment by boricj** [comment]
Silly me forgot to finish the rebase before pushing, which failed while I was AFK.

**Pull request merged**